### ISSUE TITLE
syntax highlighting in inline context blocks

### ISSUE THREAD

**Issue by hmaurer** [mattpocock/xstate-catalogue#19]
I implemented syntax highlighting for the "whole context" block (c.f. https://github.com/mattpocock/xstate-catalogue/pull/18) and I am wondering if it'd make sense to also implement it for inline context. It could look something like this:

<img width="770" alt="Screenshot 2021-04-07 at 10 42 49" src="https://user-images.githubusercontent.com/541045/113846068-016cf480-978e-11eb-9c64-d717af07c1d2.png">

Thoughts?

**Comment by mattpocock** [comment]
Yes, for sure. From a design point of view it needs a more subtle treatment. I.e. light gray etc. Maybe HighlightJS has some different themes we could try? Or create our own?

**Comment by hmaurer** [comment]
Here's another attempt:

<img width="780" alt="Screenshot 2021-04-07 at 11 46 05" src="https://user-images.githubusercontent.com/541045/113854508-d8049680-9796-11eb-8e0f-1feab3daaca7.png">

I am wondering if perhaps it'd be better to just show the name of the context key inline (e.g. `itemBeingHeld`) and show the content on hover in a popover (formatted).

**Comment by bemayr** [comment]
Nice improvement! 👏
Regarding the *hover idea*, I was already thinking of this relating to the different event payloads (https://github.com/mattpocock/xstate-catalogue/issues/15#issuecomment-814402786), but **one reason why :hover` is bad, is its problem with touch interaction**.
During research or while explaining stuff to someone I often use my convertible's touch screen and then all the **hover-features suddenly get inaccessible**.

I think that aligned with @mattpocock's idea of this project being a *transparent learning resource*, the inline context should not be larger than what can actually be nicely in-lined and everything larger should be displayed in a block. What are your opinions here?

**Comment by hmaurer** [comment]
Ah yes, you are totally right re touch interactions. This could be solved by showing the popover both on click/tap and on hover, but you'd still have the issue that it'd be hidden from a glance.

Makes sense re keeping the inline context small!.

**Comment by mattpocock** [comment]
@hmaurer Keep going, this is good.

Perhaps turning this file into a CSS module, and wrapping all the classes therein in a CSS Module would work.

https://github.com/mattpocock/xstate-catalogue/blob/master/styles/atom-one-dark.css

I.e:

```
.jsonModule {
  .hljs {
    display: block;
    overflow-x: auto;
    padding: 0.5em;
    color: theme('colors.gray.200');
  }
  // Other hljs bits
}

```

**Comment by mattpocock** [comment]
Also, I agree - inline context should be kept small. Part of the reason it's so sweet is that you can see it change on the page, without needing a hover.

**Comment by mattpocock** [comment]
@all-contributors please add @hmaurer for ideas, code

**Comment by hmaurer** [comment]
@mattpocock yeah that's what I did with the theme. In the example above I used `atom-one-light` and wrapped every class in `.hljs-inline`. I'll open a PR and we can iterate from there with the Vercel preview!

**Comment by mattpocock** [comment]
@hmaurer Gotcha, nice.

**Comment by bemayr** [comment]
I just checked the current version and really like this change, because after this got merged `context` won't look like the *sendable events* anymore!
